processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -293,7 +294,7 @@ protected IncrementalIndex(
       timeAndMetricsColumnCapabilities.put(metricDesc.getName(), metricDesc.getCapabilities());
     }
 
-    DimensionsSpec dimensionsSpec = incrementalIndexSchema.getDimensionsSpec();
+    this.dimensionsSpec = incrementalIndexSchema.getDimensionsSpec();
```

Comment:
nit: This can be moved upwards with the rest of the field initializers.

Reply:
@liran-funaro thank you for looking at my PR. I agree that could be cleaner, but want to avoid style changes if possible in this PR as it is already pretty big.